### DISCUSSION TITLE
Axum Layer Integration

### DISCUSSION THREAD

**Discussion by mcmah309** [Totodore/socketioxide#362] [Q&A] [answered]
I have an axum layer (tower) that validates and extracts information from the request and transforms it. The straight forward axum way would be to add the data to the request extension and retrieve that data in the axum handler. But how would I retrieve this data in a socketioxide handler?

**Comment by Totodore** [comment, marked as answer]
You can use the [`HttpExtension`](https://docs.rs/socketioxide/latest/socketioxide/extract/struct.HttpExtension.html) extractor to directly extract your extension data (but it will be by clone). 

Otherwise you can also use the [`req_parts`](https://docs.rs/socketioxide/latest/socketioxide/extract/struct.SocketRef.html#method.req_parts) fn on the socket ref to get the extension map.
